lib/services/local/presence.go

```diff
@@ -348,6 +348,14 @@ func (s *PresenceService) UpsertNode(ctx context.Context, server types.Server) (
 	if n := server.GetNamespace(); n != apidefaults.Namespace {
 		return nil, trace.BadParameter("cannot place node in namespace %q, custom namespaces are deprecated", n)
 	}
+	if err := utils.ValidateNodeHostname(server.GetHostname()); err != nil {
+		s.log.Warnf(
```

Comment:
Would sending a notification to all users with permissions to modify nodes be sufficient?

Reply:
If so I think doing so in a follow up PR would be ideal, as I'd like to prevent multiple identical notifications for the same node from being sent which would add to this tightly scoped PR.

Reply:
Done